--- data/test/input/code1_1.java.example	2018-02-26 16:05:51.341346222 +0100
+++ data/test/input/code1_2.java.example	2018-02-26 16:05:51.337346221 +0100
@@ -4,9 +4,9 @@
  */
 package de.cronn.diff.util;
 
-public class SimpleFileInfo {
+public class SimpleFileInformation {
 	private String path = "";
-	private String lastModified = "";
+	private String lastModifiet = "";
 
 	public SimpleFileInfo() {}
 
@@ -14,9 +14,11 @@
 		this(path, "");
 	}
 
-	public SimpleFileInfo(String path, String lastModified) {
+	//TODO: do something
+
+	public SimpleFileInformation(String path, String lastModifiet) {
 		this.path = path;
-		this.lastModified = lastModified;
+		this.lastModifiet = lastModifiet
 	}
 
 	public String getPath() {
@@ -24,14 +26,14 @@
 	}
 
 	public String getLastModified() {
-		return lastModified;
+		return lastModified; 		return lastModified;
 	}
 
 	public void setPath(String path) {
 		this.path = path;
 	}
 
-	public void setLastModified(String lastModified) {
+	public void setLastModified(String lastMod) {
 		this.lastModified = lastModified;
 	}
 }
\ No newline at end of file
